node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/SslConnectionSocketFactoryUpdater.java

```diff
@@ -1,91 +1,86 @@
+                .map(siaConfig ->
+                             new SiaIdentityProvider(
+                                     (AthenzService) AthenzIdentities.from(siaConfig.hostIdentityName()),
+                                     Paths.get(siaConfig.credentialsPath()),
+                                     new File(siaConfig.trustStoreFile())))
+                .orElse(null);
+        HostnameVerifier configServerHostnameVerifier = configServerInfo.getSiaConfig()
+                .map(siaConfig -> createHostnameVerifier(AthenzIdentities.from(siaConfig.configserverIdentityName())))
+                .orElseGet(SSLConnectionSocketFactory::getDefaultHostnameVerifier);
+        return new SslConnectionSocketFactoryUpdater(siaIdentityProvider, configServerHostnameVerifier);
     }
 
-    /** Non-private for testing only */
-    SslConnectionSocketFactoryUpdater(
-            ConfigServerInfo configServerInfo,
-            String hostname,
-            ConfigServerKeyStoreRefresherFactory refresherFactory,
-            SslConnectionSocketFactoryCreator socketFactoryCreator) {
-        this.configServerInfo = configServerInfo;
-        this.socketFactoryCreator = socketFactoryCreator;
-
-        // ConfigServerApi used to refresh the key store. Does not itself rely on a socket
-        // factory with key store, of course.
-        SSLConnectionSocketFactory socketFactoryWithoutKeyStore =
-                socketFactoryCreator.createSocketFactory(configServerInfo, Optional.empty());
-        configServerApi = ConfigServerApiImpl.createWithSocketFactory(
-                configServerInfo.getConfigServerUris(), socketFactoryWithoutKeyStore);
-
-        // If we have keystore options, we should make sure we use the keystore with the latest certificate,
-        // start the keystore refresher.
-        keyStoreRefresher = configServerInfo.getKeyStoreOptions().map(keyStoreOptions -> {
-            ConfigServerKeyStoreRefresher keyStoreRefresher = refresherFactory.create(
-                    keyStoreOptions,
-                    this::updateSslConnectionSocketFactory,
-                    configServerApi,
-                    hostname);
-
-            // Run the refresh once manually to make sure that we have a valid certificate, otherwise fail.
-            try {
-                keyStoreRefresher.refreshKeyStoreIfNeeded();
-                updateSslConnectionSocketFactory();
-            } catch (Exception e) {
-                throw new RuntimeException("Failed to acquire certificate to config server", e);
-            }
-
-            keyStoreRefresher.start();
-            return keyStoreRefresher;
-        });
+    SslConnectionSocketFactoryUpdater(SiaIdentityProvider siaIdentityProvider,
+                                      HostnameVerifier configServerHostnameVerifier) {
+        this.configServerHostnameVerifier = configServerHostnameVerifier;
+        this.sia = siaIdentityProvider;
+        if (siaIdentityProvider != null) {
```

Comment:
Express this by making siaIdentityProvider Optional.

Reply:
I follow the official recommendation that `Optional` should only be used as return parameter.

Reply:
That seems counter-productive, let's discuss offline